source_modelling/sources.py

```diff
@@ -422,6 +460,67 @@ def wgs_depth_coordinates_to_fault_coordinates(
             raise ValueError("Specified coordinates do not lie in plane")
         return np.clip(fault_local_coordinates, 0, 1)
 
+    def rrup_distance(self, point: np.ndarray) -> float:
+        """Compute RRup Distance between a fault and a point.
+
+        Parameters
+        ----------
+        point : np.ndarray
+            The point to compute distance to (in lat, lon, depth format)
+
+
+        Returns
+        -------
+        float
+            The rrup distance (in metres) between the point and the fault geometry.
+
+        Raises
+        ------
+        ValueError
+            If the optimisation routine fails to converge for the point.
```

Comment:
Isn't this only if "Degenerate line segment q -> r!" from the valueError in qcore? No longer optimisation

Reply:
Yes and even that shouldn't be thrown because it would be a bug if the line segments the plane code passed qcore were degenerate. So that documentation shouldn't exist.